src/utils/database-common/src/transactions/db_transaction_manager.rs

```diff
@@ -135,9 +135,9 @@ impl TransactionRef {
     pub fn into_maybe_transaction<DB: sqlx::Database>(
         self,
     ) -> Option<sqlx::Transaction<'static, DB>> {
-        Arc::try_unwrap(self.inner)
-            .ok()
-            .and_then(|m| m.into_inner().maybe_transaction)
+        let m = Arc::try_unwrap(self.inner).unwrap().into_inner();
```

Comment:
I'd suggest replacing `unwrap()` with `expect("..")` to add a good message explaining a case with "stuck" transaction reference